src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2690,11 +2689,16 @@ mesh::generate_index(const Node &mesh,
             }
             else if(matset.has_child("volume_fractions"))
             {
+                // we don't have material_map (node with names to ids)
+                // so mapping is implied from node order, construct
+                // an actual map that follows the implicit order
                 NodeConstIterator mats_itr = matset["volume_fractions"].children();
+                index_t mat_id = 0;
                 while(mats_itr.has_next())
                 {
                     mats_itr.next();
-                    idx_matset["materials"][mats_itr.name()];
+                    idx_matset["material_map"][mats_itr.name()] = mat_id;
```

Comment:
Unless I'm missing something, this could be simplified by using `mats_itr.index()` instead.